### PR TITLE
Feature/add workflow completion listener al

### PR DESCRIPTION
**Pull request by alexnetflix**
Adding a listener for when workflow gets to a COMPLETED or TERMINATED state.
This listener could be redefined in projects that use or inherit conductor library with some custom implementation as needed.

WorkflowDefinition has a flag to trigger these listeners so that we do not trigger these for every workflow.

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowStatusListenerStub.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+/**
+ * Stub listener default implementation
+ */
+public class WorkflowStatusListenerStub implements WorkflowStatusListener {
+
+    private static final Logger LOG = LoggerFactory.getLogger(WorkflowStatusListenerStub.class);
+
+    @Override
+    public void onWorkflowCompleted(Workflow workflow) {
+        LOG.debug("Workflow {} is completed", workflow.toString());
```
Comment (cyzhao) [inline core/src/main/java/com/netflix/conductor/core/execution/WorkflowStatusListenerStub.java]: only log workflowId

File: common/src/main/java/com/netflix/conductor/common/metadata/workflow/WorkflowDef.java [hunk 2]
```diff
@@ -66,6 +66,9 @@
 	@ProtoField(id = 9)
 	private boolean restartable = true;
 
+	@ProtoField(id = 10)
+	private boolean terminalWorkflowListenerEnabled = false;
```
Comment (cyzhao) [inline common/src/main/java/com/netflix/conductor/common/metadata/workflow/WorkflowDef.java]: consider naming it workflowStatusListenerEnabled to be consistent with interface name, also be generic as we should also notify on other status changes, e.g. when a workflow is restarted.

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowStatusListenerStub.java (hunk 1 above)
Comment (alexnetflix) [reply core/src/main/java/com/netflix/conductor/core/execution/WorkflowStatusListenerStub.java]: done

File: common/src/main/java/com/netflix/conductor/common/metadata/workflow/WorkflowDef.java (hunk 2 above)
Comment (alexnetflix) [reply common/src/main/java/com/netflix/conductor/common/metadata/workflow/WorkflowDef.java]: renamed

**Pull request merged**